Adds initial commit for github client api

NunoFrRibeiro opened the pull request:
Addresses: #12

NunoFrRibeiro committed:
Adds initial commit for github client api

Signed-off-by: NunoFrRibeiro <nunofrribeiro@hotmail.com>

NunoFrRibeiro committed:
Add first iteration of github_client

Signed-off-by: NunoFrRibeiro <nunofrribeiro@hotmail.com>

NunoFrRibeiro committed:
chore(github_client): adds more fields do request

Addresses: #12

Adds more fields to the GraphQL API request
Parses the result as json

Signed-off-by: NunoFrRibeiro <nunofrribeiro@hotmail.com>

NunoFrRibeiro committed:
chore(github_client): Adds Unit testing

Addresses: #12

Signed-off-by: NunoFrRibeiro <nunofrribeiro@hotmail.com>

NunoFrRibeiro committed:
chore(github_client): Add more unit testing

Addresses: #12

Signed-off-by: NunoFrRibeiro <nunofrribeiro@hotmail.com>

NunoFrRibeiro committed:
feat(github-client): refactor unit tests

addresses: #12

NunoFrRibeiro committed:
feat(github-client): refactor pyproject

Addresses: #12

Signed-off-by: NunoFrRibeiro <nunofrribeiro@hotmail.com>

NunoFrRibeiro committed:
feat(github-handler): move graphql module

Addresses #12

Signed-off-by: NunoFrRibeiro <nunofrribeiro@hotmail.com>

NunoFrRibeiro committed:
feat(github_client): improve logic to fetch data

Addresses: #12

Signed-off-by: NunoFrRibeiro <nunofrribeiro@hotmail.com>

NunoFrRibeiro committed:
feat(github_client): refactor tests

Signed-off-by: NunoFrRibeiro <nunofrribeiro@hotmail.com>

rodrigofronza commented on src/utils/utils.py:
```diff
@@ -0,0 +1,84 @@
+
+
+class Utils:
+    def format_json_outputself(self, runs):
```
```suggestion
    @staticmethod
    def format_json_outputself(runs):
```
Doesn't seems to need self.

NunoFrRibeiro replied:
done as suggested

rodrigofronza commented on src/utils/utils.py:
```diff
@@ -0,0 +1,84 @@
+                        "status": job.get("status"),
+                        "started_at": job.get("startedAt"),
+                        "completed_at": job.get("completedAt"),
+                        "duration": self.__format_duration(
```
```suggestion
                        "duration": Utils.__format_duration(
```

NunoFrRibeiro replied:
done as suggested

rodrigofronza commented on src/utils/utils.py:
```diff
@@ -0,0 +1,84 @@
+
+        print(json.dumps(output, indent=2))
+
+    def __format_duration(self, started_at, completed_at):
```
```suggestion
    @staticmethod
    def __format_duration(started_at, completed_at):
```
Same here.

NunoFrRibeiro replied:
done as suggested

rodrigofronza commented on src/github_client.py:
```diff
@@ -0,0 +1,128 @@
+                        workflow_runs.append(simplified_run)
+
+        workflow_runs.sort(key=lambda x: x.get("created_at", ""), reverse=True)
+        # workflow_runs = workflow_runs[:10]
```
Remove this comment.

NunoFrRibeiro replied:
done as suggested

rodrigofronza commented on src/utils/utils.py:
```diff
@@ -0,0 +1,84 @@
+
+
+utils = Utils()
+format_json_output = utils.format_json_outputself
```
```suggestion
format_json_output = Utils.format_json_outputself
```

NunoFrRibeiro replied:
done as suggested

rodrigofronza commented on src/main.py:
```diff
@@ -1,6 +1,56 @@
+        runs = workflow_runs.get_last_workflow_runs()
+    except GitHubClientError as exc:
+        logger.error(f"Error: {exc}")
+        return 1
```
What if we raise something instead of `return 1`

NunoFrRibeiro replied:
for now it's raising a ValueError exception.
Does it make sense to raise any other exception?

rodrigofronza commented on src/main.py:
```diff
@@ -1,6 +1,56 @@
+
+    print(json.dumps(output, indent=2))
+
+    return 0
```
This is also probably not needed.

NunoFrRibeiro replied:
removed both `print` and the `return`

NunoFrRibeiro committed:
review(github_client): address code review

- makes methods inside the `utils.py` static and removes self from
  arguments
- removes `return` statement from `main.py` if script fails it just
  raises a ValueError exception

Address: #12

Signed-off-by: NunoFrRibeiro <nunofrribeiro@hotmail.com>

NunoFrRibeiro committed:
review(github_client): address code review

- makes methods inside the `utils.py` static and removes self from
  arguments
- removes `return` statement from `main.py` if script fails it just
  raises a ValueError exception

Address: #12

Signed-off-by: NunoFrRibeiro <nunofrribeiro@hotmail.com>

rodrigofronza commented on src/utils/utils.py:
```diff
@@ -0,0 +1,85 @@
+
+class Utils:
+    @staticmethod
+    def format_json_outputself(runs):
```
```suggestion
    def format_json_output(runs):
```

NunoFrRibeiro replied:
addressed!

rodrigofronza commented on src/utils/utils.py:
```diff
@@ -0,0 +1,85 @@
+            return "N/A"
+
+
+format_json_output = Utils.format_json_outputself
```
```suggestion
format_json_output = Utils.format_json_output
```

NunoFrRibeiro replied:
addressed!
